jwst/pipeline/calwebb_spec3.py

```diff
@@ -75,17 +76,26 @@ def process(self, input):
         output_file = input_models.meta.asn_table.products[0].name
         self.output_file = output_file
 
-        # Split the inputs into science and background
-        science_data, background_data = split_container(input_models)
+        # Find all the member types in the product
+        members_by_type = defaultdict(list)
+        product = input_models.meta.asn_table.products[0].instance
+        for member in product['members']:
+            members_by_type[member['exptype'].lower()].append(member['expname'])
 
         # If background data are present, call the master background step
-        if len(background_data) > 0:
+        if len(members_by_type['background']) > 0:

```

Comment:
```
if members_by_type['background']:
```
is the pythonic, "truthy" check.  If the list has no length, it evaluates to `False`.

Reply:
Done